assets/src/components/site-scan-context-provider/index.js

```diff
@@ -265,9 +291,29 @@ export function SiteScanContextProvider( {
 	const previousDidSaveOptions = usePrevious( didSaveOptions );
 	useEffect( () => {
 		if ( ! previousDidSaveOptions && didSaveOptions ) {
+			cancelSiteScan();
 			fetchScannableUrls();
 		}
-	}, [ didSaveOptions, fetchScannableUrls, previousDidSaveOptions ] );
+	}, [ cancelSiteScan, didSaveOptions, fetchScannableUrls, previousDidSaveOptions ] );
+
+	/**
+	 * Delay concurrent validation requests.
+	 */
+	const [ canRequestUrlValidation, setCanRequestUrlValidation ] = useState( true );
```

Comment:
Is `canRequestUrlValidation` the best name? This sounds like a permissions thing for whether the user has the capability to scan. Should rather be something like `isCurrentlyRequestingValidation` or `shouldDelayValidationRequest`?

Reply:
That's a good point. Addressed in 213e4f7.